### PR TITLE
Resolving a number of issues

### PR DESCRIPTION
**Pull request by spaulaus**
The review process has become an impediment to some work. I have forked my own copy of the software so that I can quickly fix issues that are coupled.

# Resolved Issues
- #20
- #29
- #55
- #60
- #61
- #68
- #92
- #103 
- #105 
- #117
- #118
- #119
- #157 
- #228
- #233 
- #235
- #236 
- #237 
- #238
- #244
- #249
- #250 
- #251 
- #252
- #262 
- #271
- #272 
- #273 
- #274 
- #275 
- #276 

# Additional Updates
- RootScanner is now RootInterface. It lives in Resources and provides all programs an interface to a ROOT canvas.
- Reformatting code base to remove tabs, adhere to line lengths, and bringing things more in line with style guidelines.
- Corrected where `UtkUnpakcer::eventWidth_` is set so that it's set properly before any events are built.
- Removed QdcCompression from Globals and put it into the VandleProcessor constructor. This really shouldn't be used outside of the VandleProcessor anyway.

### CODE REVIEW

File: Analysis/Resources/include/ChannelConfiguration.hpp [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    ChannelConfiguration() : location_(9999), subtype_(""), type_("") {}
+
+    ~ChannelConfiguration() {}
+
+    ///Constructor taking arguments and setting values, this constructor is used to maintain functionality with the
+    /// previous Identifier class.
+    ///@param [in] atype : the type to set
+    ///@param [in] subType : the subType to set
+    ///@param [in] loc : the location to set
+    ChannelConfiguration(const std::string &atype, const std::string &subType, const unsigned int &loc) {
```
Comment (ksmith0) [inline Analysis/Resources/include/ChannelConfiguration.hpp]: Rewrite to use initializer list.

File: Analysis/Resources/include/GslFitter.hpp [hunk 2]
```diff
@@ -23,15 +23,6 @@ class GslFitter : public TimingDriver {
     ///Default Destructor
     ~GslFitter() {}
 
-    /// @return the amplitude from the GSL fit
```
Comment (ksmith0) [inline Analysis/Resources/include/GslFitter.hpp:26]: Why did this get removed?

File: Analysis/Resources/include/ChannelConfiguration.hpp (hunk 1 above)
Comment (spaulaus) [reply Analysis/Resources/include/ChannelConfiguration.hpp]: Fixed in the next push

File: Analysis/Resources/include/GslFitter.hpp (hunk 2 above)
Comment (spaulaus) [reply Analysis/Resources/include/GslFitter.hpp:26]: These were moved to TimingDriver for enhanced compatibility.

Comment (spaulaus) [reply Analysis/Resources/include/GslFitter.hpp:26]: ...and extend functionality.

File: Analysis/ScanLibraries/source/ScanInterface.cpp [hunk 3]
```diff
@@ -978,26 +977,22 @@ bool ScanInterface::Setup(int argc, char *argv[],
         }
     }//while
 
-    //We check that the unpacker object has been set.
-    unpacker_ = unpacker;
-    if (!unpacker_)
+    if (!unpacker)
         throw invalid_argument("ScanInterface::Setup - The Unpacker object has not been set properly.");
+    else
```
Comment (ksmith0) [inline Analysis/ScanLibraries/source/ScanInterface.cpp]: You don't need the else here.

File: Analysis/ScanLibraries/source/ScanInterface.cpp [hunk 4]
```diff
@@ -978,26 +977,22 @@ bool ScanInterface::Setup(int argc, char *argv[],
...
+        unpacker_ = unpacker;
```
Comment (ksmith0) [inline Analysis/ScanLibraries/source/ScanInterface.cpp]: You don't need the else here.

**Comment by spaulaus** [comment]
> What if Minuit2 isn't installed? Can we do this or the gErrorLevel thing from hist if it is not?

I changed it so that if Minuit2 isn't returned from root-config then there's some pre-processor defines in there. That was simple and efficient way of doing this. The pre-processor define only covers a few lines of code, not whole swaths.

**Comment by ksmith0** [comment]
That doesn't answer the issue with Minuit throwing up error messages everywhere. I believe histScanner was making use of `gErrorLevel` to suppress messages like this and we should be able to do the same with scope.

**Comment by tking53** [comment]
Im curious why we wouldn't want the QDC compression available to other processors. Could it not be useful in any processor that deals with traces (any of the Beta or Liquid or maybe future HAGRiD processors), or are you thinking that we would have a different one per processor? 36f2201

**Comment by tking53** [comment]
Also on behalf of Miguel and Robert I'm going to request this.

Can we have type:subtype defaults for the new Calibration nodes? They believe that the CFG is already too long.  
I see that the configuration default to Vandle:Medium settings but it might be nice to have the ability to set defaults for other type:subtype detectors with traces; like Vandle:Small or LaBr3 or whatever other large combined array robert comes up with in the future (VANDLE+NEXT has been floated around)

I believe that might ease the transition after this massive PR

https://github.com/spaulaus/paass/blob/5fb258d729f111d5cfe198aead033b89a0d49dda/Analysis/Resources/include/DefaultConfigurationValues.hpp#L10-L19

File: Analysis/Utilities/RootScanner/source/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+root_generate_dictionary(HistScannerDictionary HistScannerChanData.hpp
+        LINKDEF HistScannerLinkDef.h)
+add_executable(rootscan HistUnpacker.cpp HistScanner.cpp HistScannerChanData.cpp
+        HistScannerDictionary.cxx hist.cpp)
+target_link_libraries(rootscan PaassScanStatic ResourceStatic PugixmlStatic
+        PaassResourceStatic ${ROOT_LIBRARIES} -lTreePlayer)
```
Comment (ksmith0) [inline Analysis/Utilities/RootScanner/source/CMakeLists.txt:6]: Check this linking to TreePlayer.

Comment (ksmith0) [reply Analysis/Utilities/RootScanner/source/CMakeLists.txt:6]: ROOT suggests using the following:

```
find_package(ROOT REQUIRED COMPONENTS TreePlayer)
```

**Review by ksmith0 — APPROVED** [review]

**Pull request merged**